plugin/core/sessions.py

```diff
@@ -1136,6 +1147,10 @@ def m_client_registerCapability(self, params: Any, request_id: Any) -> None:
                     # Inform only after the response is sent, otherwise we might start doing requests for capabilities
                     # which are technically not yet done registering.
                     sublime.set_timeout_async(inform)
+            if self._plugin:
+                inform = functools.partial(
+                    self._plugin.on_register_capability_async, registration_id, capability_path, options)
+                sublime.set_timeout_async(inform)
```

Comment:
Not sure now why I've decided to do it from `set_timeouta_async`. Maybe not to delay the response but it shouldn't really be needed, now that I think of it.

Reply:
Note, 
The `client/registerCapability` is sent from the server to the client, 
and the response for this request is void. (same for `client/unregisterCapability`)

Reply:
OK, but I'm not sure I understand the point of this note. :)

Reply:
You wrote `"Maybe not to delay the response"`, 
So I took a look at the LSP spec and saw that the response is `void`. So I also started thinking that `set_timeout_async` is maybe not needed. 
... Hope it makes more sense now, if not just skip these messages :)

Reply:
I see.
What I meant is that the blocking code in the plugin could delay sending of the void response. Probably either way is fine though.

Reply:
This looks okay to me. We have to do it via `set_timeout_async` because otherwise the response to this `client/registerCapability` has not yet been sent back to the language server.

Reply:
RIght. Probably that is what I was thinking too.